Use masterbar height var instead of explicit values

cpapazoglou opened the pull request:
There are many parts that depend on masterbar height and they are hardcoded, making the testing and implementation of nav unification difficult. This PR attempts to alleviate this by using a global var for masterbar

#### Changes proposed in this Pull Request

* introduces `--masterbar-height: 46px` var
* adds `box-sizing: border-box` so that the 1px border is calculated to masterbars height
* replaces all instances of elements being `fixed` with `top: 47px` to `var (--masterbar-height )
* edits `apps/notifications/style.scss` accordingly by removing not needed rules anymore

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Repeat below steps with both the flag `?flags=nav-unification` and without it
* load calypso check that sidebar, notifications panel etc appear correctly both in desktop and responsive

**Known Issue** with `?flags=nav-unification` 
![](https://cln.sh/Qkj6NX+)
This type of gaps will be resolved once we move 
```
@media only screen and ( min-width: 782px ) {
  --masterbar-height: 32px;
}
```
to the `:root`

getdave commented on apps/notifications/style.scss:
```diff
@@ -6,7 +6,7 @@
 #wpnc-panel {
 	position: fixed;
-	top: 47px;
+	top: var( --masterbar-height );
```
I wonder if changes to apps would be better as an Issue. Otherwise you're going to get caught up in building and deploying the Notifications app (cc @marekhrabe who knows about this) instead of focusing on shipping Calypso UI code.

cpapazoglou replied:
We will eventually need to touch this file. Not sure why we need to deploy the Notifications app right now? I think it can be deployed when someone needs it updated. As far as calypso is concerned, there is no need for deploy.

marekhrabe replied:
This is interesting because we might need to alter the behavior of the notifications app in iframe based on the flag of the site it's being embedded it. This might get pretty complex I'm afraid 😬

cpapazoglou replied:
@marekhrabe Thanks for providing PCYsg-elI-p2, now I got it! Eventually, I checked the frontend notifications 
![](https://cln.sh/qhKwnU+)
and they don't depend `#wpnc-panel`. In fact `#wpnc-panel` only exists in calypso context. Notifications in the frontend are working already great and this change will not affect them.

getdave commented on client/assets/stylesheets/_nav-unification.scss:
```diff
@@ -57,7 +48,7 @@
 		height: 24px;
 		margin: 4px 8px;
-		border-radius: 5px;
+		border-radius: 2px;
```
What's this change for?

cpapazoglou replied:
I tried to comply with https://github.com/Automattic/wp-calypso/blob/12df4ecc07993c51c549616de6a1c93ce7d95077/.stylelintrc#L6

But, I just thought that this has to comply with `wp-admin` so I reverted it https://github.com/Automattic/wp-calypso/pull/46035/commits/7014168217b5798fc5d03b54442b33080e639e6e

Good catch, thanks.

getdave commented on client/assets/stylesheets/_nav-unification.scss:
```diff
@@ -11,38 +11,29 @@
 .is-nav-unification {
+	// breakpoint used in wp-admin
+	@media only screen and ( min-width: 782px ) {
+		--masterbar-height: 32px;
```
I think you should note in a comment that this overides  the original which is set in `client/assets/stylesheets/shared/_variables.scss`

cpapazoglou replied:
Added https://github.com/Automattic/wp-calypso/pull/46035/commits/bb7a1438c65165f95a5cc0342ec7f77d7ed01ca7

getdave commented on client/layout/masterbar/style.scss:
```diff
@@ -17,6 +16,7 @@ $autobar-height: 20px;
 	z-index: z-index( 'root', '.masterbar' );
 	-webkit-font-smoothing: subpixel-antialiased;
 	transition: transform 0.2s ease-out;
+	box-sizing: border-box;
```
I'm worried about the wider impact of this change. `box-sizing` has the potential to affect styles that apply to the horizontal margins. If folks have relied on this property for some time then there might be unintended consequences of making this switch.

Personally I'd try and limit the scope of this change to just making a 1:1 swap between the Sass and CSS vars and avoid amending this property.

cpapazoglou replied:
> making a 1:1 swap between the Sass and CSS vars

In that case we would need to replace all `var(--masterbar-height)` with `calc( var(--masterbar-height) + 1px)` where 1px is the border height. Thus introducing some technical debt.

The new masterbar (for the nav-unification) already [didn't use `border` but `box-shadow` instead](https://github.com/Automattic/wp-calypso/pull/46035/files#diff-24092f405ff9cb89da5651239314443fL33-L37) which I changed to align better with the current code. That means that if we are not introducing `box-sizing: border-box;` we will inevitably may have  _unintended consequences_ once the nav-unification goes live.

Since the diff is only 1px, I would prefer avoiding some technical debt and address any _unintended consequences_ as soon as possible.

getdave replied:
You seem pretty confident and to be fair I cannot see any problems on testing so I'm inclined to agree with you. Nonetheless, let's continue to be aware of the potential for knock-on effects. Hopefully, they will be minimal 🤞 .

getdave reviewed (changes requested):
Thanks for the PR. I can see the value in making the height value dynamic as a var.

I have concerns about 2 things:

1. Amending anything in `apps/` - feels like additional overhead.
2. Changing the `box-sizing` value as this could have unintended knock on effects.

I did test this and I couldn't see any regressions. That said, I can easily imagine some creeping through unnoticed.

My instinct is that we should descope this PR to:

1. Avoid amending code in `apps/`.
2. Avoid changing the value of the var we are replacing (eg: `47px` which now becomes `46px` in the var due to `border-box` change). Keep the value the same and avoid `border-box` change even if that adds more overhead in the short term.

Let me know what you think.

Nice work though.

cpapazoglou commented:
> Thanks for the PR. I can see the value in making the height value dynamic as a var.
> 
> I have concerns about 2 things:
> 
> 1. Amending anything in `apps/` - feels like additional overhead. Note also you might need to "build" the app in order to see the changes reflected. Would also require testing in Jetpack connected site where the Notifications app is used.
> 2. Changing the `box-sizing` value as this could have unintended knock on effects.
> 
> I did test this and I couldn't see any regressions. That said, I can easily imagine some creeping through unnoticed.
> 
> My instinct is that we should descope this PR to:
> 
> 1. Avoid amending code in `apps/`.
> 2. Avoid changing the value of the var we are replacing (eg: `47px` which now becomes `46px` in the var due to `border-box` change). Keep the value the same and avoid `border-box` change even if that adds more overhead in the short term.
> 
> Let me know what you think.
> 
> Nice work though.

Thanks @getdave for your prompt review! This PR surely is a tough one! I addressed your points independently so let's continue there.

marekhrabe commented:
I'm not too worried about this change in `apps/` dir, I think we can pull it off! I'll test it and confirm things are working okay. Instructions PCYsg-elI-p2

getdave commented:
> I'm not too worried about this change in `apps/` dir, I think we can pull it off! I'll test it and confirm things are working okay. Instructions [PCYsg-elI-p2](https://href.li/?PCYsg-elI-p2)

If you think it's doable then that's cool. I just thought it's one less dependency and thing to do if we just split out this one change into a separate PR. You can still work on the notifications app, but it won't be a dependency of this PR which will allow it to merge sooner.

Happy to leave it with you both 🤝

getdave reviewed (approved):
Having tested I cannot see any immediate issues with this.

As you know I had/have concerns about the `box-sizing` change, but I can't immediately _see_ any issues and you seem confident we can handle knock-on effects.

Therefore it's a 👍 from me. Nice work.

PS: if changes to Notifications hold you up any more than I would pull them into a separate PR and deal with them in isolation.
